pkg/api/team_members.go

```diff
@@ -153,10 +156,20 @@ func (hs *HTTPServer) RemoveTeamMember(c *models.ReqContext) response.Response {
 	return response.Success("Team Member removed")
 }
 
-// addTeamMember adds a team member.
+// addOrUpdateTeamMember adds or updates a team member.
 //
 // Stubbable by tests.
-var addTeamMember = func(sqlStore *sqlstore.SQLStore, userID, orgID, teamID int64, isExternal bool,
+var addOrUpdateTeamMember = func(resourcePermissionService *resourcepermissions.Service, userID, orgID, teamID int64, isExternal bool,
 	permission models.PermissionType) error {
-	return sqlStore.AddTeamMember(userID, orgID, teamID, isExternal, permission)
+	permissionString := permission.String()
+	// Team member permission is 0, which maps to an empty string.
+	// However, we want the team permission service to display "Member" for team members. This is a hack to make it work.
+	if permissionString == "" {
+		permissionString = "Member"
+	}
+	actions := resourcePermissionService.MapPermission(permissionString)
+	teamIDString := strconv.FormatInt(teamID, 10)
+	_, err := resourcePermissionService.SetUserPermission(context.TODO(), orgID, userID, teamIDString, actions)
```

Comment:
nit: seems this error from resourcePermissionService is also returned in a few other places, you may want to wrap it before returning